src/sage/rings/burnside.py

```diff
@@ -0,0 +1,273 @@
+from sage.misc.cachefunc import cached_method
+from sage.structure.unique_representation import UniqueRepresentation
+from sage.structure.parent import Parent
+from sage.structure.element import Element
+from sage.rings.integer_ring import ZZ
+from sage.structure.formal_sum import FormalSum
+from sage.categories.sets_cat import cartesian_product, Sets
+from sage.groups.perm_gps.permgroup import PermutationGroup
+from sage.libs.gap.libgap import libgap
+from sage.categories.algebras import Algebras
+from sage.combinat.free_module import CombinatorialFreeModule
+
+def _is_conjugate(G, H1, H2):
+    return libgap.eval('fail') != libgap.RepresentativeAction(G, H1, H2)
+
+class ConjugacyClassOfSubgroups(Element):
+    def __init__(self, parent, C):
+        Element.__init__(self, parent)
+        self._C = C
+
+    def __hash__(self):
+        return hash(self._C)
+
+    def _repr_(self):
+        name = self.parent()._names.get(self._C, None)
+        if name is None:
+            return repr(self._C.gens())
+        else:
+            return name
+
+    def __le__(self, other):
+        return libgap.eval('fail') != libgap.ContainedConjugates(self.parent()._G, other._C, self._C, True)
+
+    def __eq__(self, other):
+        return _is_conjugate(self.parent()._G, self._C, other._C)
+
+class ConjugacyClassesOfSubgroups(Parent):
+    def __init__(self, G):
+        self._G = G
+        self._cache = dict() # invariant to subgroups
+        self._names = dict() # dictionary mapping subgroups to names
+        Parent.__init__(self, category=Sets().Finite())
```

Comment:
```suggestion
        Parent.__init__(self, category=FiniteEnumeratedSets())
```
allows us to get rid of our custom `gens` method